src/adapter/src/catalog.rs

```diff
@@ -3185,16 +3190,12 @@ impl Catalog {
             let serialized_item = Self::serialize_item(item);
             tx.insert_item(id, schema_id, &name, serialized_item)?;
         }
-        tx.update_system_object_mappings(
-            migration_metadata
-                .migrated_system_object_mappings
-                .drain()
-                .collect(),
-        )?;
+        tx.update_system_object_mappings(std::mem::take(
+            &mut migration_metadata.migrated_system_object_mappings,
+        ))?;
```

Comment:
`BTreeMap` does not provide a `drain` method. I replaced uses of `HashMap::drain` with `std::mem::take(map).into_iter()` instead.

(Or in this case just `std::mem::take`, as the iterate-and-collect step was superfluous.)